src/ess/reduce/live/raw.py

```diff
@@ -343,8 +377,32 @@ def get(self, window: int | None = None) -> sc.DataArray:
                 data += self._history['window', 0 : self._current].sum('window')
         return data
 
+    def add_events(self, data: sc.DataArray) -> None:
+        """
+        Add counts in the form of events grouped by pixel ID.
+
+        Parameters
+        ----------
+        data:
+            Events grouped by pixel ID, given by binned data.
+        """
+        counts = data.bins.size().to(dtype='int32', copy=False)
+        counts.unit = 'counts'
+        self._add_counts(counts)
+
     def add_counts(self, data: Sequence[int]) -> None:
```

Comment:
This will not be used by Beamlime anymore, keeping it for backwards compatibility.